api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -424,6 +424,12 @@ message HttpConnectionManager {
   // Note that Envoy does not perform
   // `case normalization <https://tools.ietf.org/html/rfc3986#section-6.2.2.1>`
   google.protobuf.BoolValue normalize_path = 30;
+
+  // If present and set to true, the connection manager will validate the values of each HTTP
+  // header, returning a 400 response if any of the headers contain an invalid
+  // value. See `Header Fields <https://tools.ietf.org/html/rfc7230#section-3.2>`
+  // for more details regarding the validity of HTTP headers.
+  bool validate_headers = 33;
```

Comment:
We can sanity check with @mattklein123 , but if this is making Envoy more standards compliant I'd lean towards making it a permanent change (off by default, and then see if on by default works for people)

If so I think instead of making changes to the Envoy API, we should just runtime guard per
https://github.com/envoyproxy/envoy/blob/master/CONTRIBUTING.md#runtime-guarding

Reply:
runtime guard for this SGTM :+1:

Reply:
Okay.  Just so I understand correctly, we'd be making this new behavior:

- a runtime-guarded option _only_, and removing the option it from the connection manager proto, as I have it currently.
- off by default

I'm basing that on the comments in [the original issue](https://github.com/envoyproxy/envoy/issues/7061) about making it opt-in.

I was a little confused by this verbiage from [the runtime guarding docs](https://github.com/envoyproxy/envoy/blob/5ee70cb19ca3bb6d817bae497b63c2a250d2d16b/CONTRIBUTING.md#runtime-guarding):
> Generally all runtime guarded features will be set true when a release is cut)